src/js/components/ClusterPicker.js

```diff
@@ -25,13 +29,37 @@ const ClusterPickerWrapper = (styled.div`
 `: ComponentType<*>)
 
 export default function ClusterPicker() {
-  const current = useSelector(Current.mustGetConnection)
+  const dispatch = useDispatch()
+  const current = useSelector(Current.getConnection)
+  const clusters = useSelector(Clusters.all)
+
+  const template = clusters.map((c) => {
+    const isCurrent = c.id === current.id
+    return {
+      type: "checkbox",
+      label: c.id,
+      checked: isCurrent,
+      click: () => !isCurrent && dispatch(setConnection(c))
+    }
+  })
+
+  template.push(
+    {type: "separator"},
+    {
+      label: "+ New Connection",
+      click: () => dispatch(Current.setConnectionId(""))
+    }
+  )
+
+  const openMenu = usePopupMenu(template)
+
+  const onContextMenu = (e) => {
+    electronIsDev && openMenu(e.currentTarget)
+  }
 
   return (
-    <ClusterPickerWrapper>
-      <label>
-        {current.host}:{current.port}
-      </label>
+    <ClusterPickerWrapper onContextMenu={onContextMenu}>
+      <label>{current.id}</label>
```

Comment:
Right now the connection id happens to the be the host:port combo, but it seems unreliable to assume that will be the case in the long-term. I'd suggest that when we actually need the host, we call a function with a name like getHost. 

For example, when we need to construct the zealot client, I think it's more clear to have a function called `Current.getConnectionHost` instead of `Current.getConnectionId`, even thought at the moment the result is the same.

The same goes for the line above. I'd suggest keeping the {host}:{port} join or making a little function somewhere that does it.